tests/napari/test_training.py

```diff
@@ -86,9 +90,101 @@ def test_run_with_virtual_yaml_files(get_training_widget):
         expected_network_args.trained_model = None
         expected_network_args.model_weights = None
 
-        run_training.assert_called_once_with(
+        mock_training_worker.assert_called_once_with(
             expected_training_args,
             expected_network_args,
             expected_optional_training_args,
             expected_misc_args,
         )
+
+        mock_worker_instance.start.assert_called_once()
+
+
+@pytest.fixture
+def training_worker_inputs():
+    """Fixture to provide standard inputs for TrainingWorker tests."""
+    return {
+        "training_inputs": TrainingDataInputs(),
+        "network_inputs": OptionalNetworkInputs(),
+        "training_options": OptionalTrainingInputs(),
+        "misc_inputs": MiscTrainingInputs(),
+    }
+
+
+def test_training_worker(training_worker_inputs):
+    """Test TrainingWorker initialization and progress bar callback."""
+
+    worker = TrainingWorker(
+        training_worker_inputs["training_inputs"],
+        training_worker_inputs["network_inputs"],
+        training_worker_inputs["training_options"],
+        training_worker_inputs["misc_inputs"],
+    )
+
+    assert (
+        worker.training_data_inputs
+        == training_worker_inputs["training_inputs"]
+    )
+    assert (
+        worker.optional_network_inputs
+        == training_worker_inputs["network_inputs"]
+    )
+    assert (
+        worker.optional_training_inputs
+        == training_worker_inputs["training_options"]
+    )
+    assert worker.misc_training_inputs == training_worker_inputs["misc_inputs"]
+
+    progress_bar = ProgressBar()
+    worker.connect_progress_bar_callback(progress_bar)
+
+    worker.signals.update_progress.emit("Test", 10, 5)
```

Comment:
Is there a way we could test on the actual progress bar of the widget (just wondering)?

Reply:
I’ve added `test_widget_progress_bar()` to verify the progress bar updates during training. It’s currently marked `xfail` due to #443, but we’ll remove that once it’s resolved.